src/test/java/com/uid2/shared/store/RotatingServiceLinkStoreTest.java

```diff
@@ -0,0 +1,93 @@
+package com.uid2.shared.store;
+
+import com.uid2.shared.cloud.ICloudStorage;
+import com.uid2.shared.model.ServiceLink;
+import com.uid2.shared.store.reader.RotatingServiceLinkStore;
+import com.uid2.shared.store.scope.GlobalScope;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+
+import static com.uid2.shared.TestUtilites.makeInputStream;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+public class RotatingServiceLinkStoreTest {
+
+    private AutoCloseable mocks;
+    @Mock
+    ICloudStorage cloudStorage;
+    private RotatingServiceLinkStore serviceLinkStore;
+
+    @BeforeEach
+    public void setup() {
+        mocks = MockitoAnnotations.openMocks(this);
+        serviceLinkStore = new RotatingServiceLinkStore(cloudStorage, new GlobalScope(new CloudPath("metadata")));
+    }
+
+    @AfterEach
+    public void teardown() throws Exception {
+        mocks.close();
+    }
+
+    private JsonObject makeMetadata(String location) {
+        JsonObject metadata = new JsonObject();
+        JsonObject serviceLinks = new JsonObject();
+        serviceLinks.put("location", location);
+        metadata.put("service_links", serviceLinks);
+        return metadata;
+    }
+
+    private ServiceLink addServiceLink(JsonArray content, String linkId, int serviceId, int siteId, String name) {
+
+        ServiceLink l = new ServiceLink(linkId, serviceId, siteId, name);
+
+
+        JsonObject service = new JsonObject();
```

Comment:
You can use the `mapFrom()` method to create the JsonObject from the class object.
`JsonObject service = JsonObject.mapFrom(l);`
Just a suggestion :)